src/tests/functions/Add-DscResourceModule.Tests.ps1

```diff
@@ -0,0 +1,63 @@
+Describe 'Vendoring a DSC module' {
+  InModuleScope puppet.dsc {
+    Context 'When only -Name is specified' {
+      Mock New-Item -Verifiable
+      Mock Save-Module -Verifiable
+      Mock Test-Path { $false } -ParameterFilter { $path -eq './tmp' }
+      Mock Get-CanonicalPath { './tmp' }
+      Mock Test-Path { $false } -ParameterFilter { $path -eq './tmp_tmp' }
+      Mock Get-ChildItem
+      Mock Move-Item
+      Mock Remove-Item
+
+      Add-DscResourceModule -Name SomeDSCModule
+
+      It 'downloads the latest version of the module to ./tmp' {
+        Assert-VerifiableMock
+        Assert-MockCalled New-Item -Times 1 -ParameterFilter { ($path -eq './tmp') -and ($ItemType -eq 'Directory') }
+        Assert-MockCalled New-Item -Times 1 -ParameterFilter { ($path -eq './tmp_tmp') -and ($ItemType -eq 'Directory') }
+        Assert-MockCalled Save-Module -Times 1 -Debug -ParameterFilter {
+          ($Name -eq 'SomeDSCModule') `
+            -and ($path -eq './tmp_tmp') `
+            -and ($RequiredVersion -eq '')
+        }
+        Assert-MockCalled Remove-Item -Times 1 -ParameterFilter { $path -eq './tmp_tmp' }
+      }
+
+      It 'does not leave litter behind' {
+        Get-ChildItem . -filter tmp | Should -BeNullOrEmpty
+        Get-ChildItem . -filter tmp_tmp | Should -BeNullOrEmpty
+      }
+    }
+
+    Context 'With -Name, -Path and -RequiredVersion specified' {
+      Mock New-Item -Verifiable
+      Mock Save-Module -Verifiable
+      Mock Test-Path { $false } -ParameterFilter { $path -eq 'C:\target\' }
+      Mock Get-CanonicalPath { 'C:\target' } -ParameterFilter { $path -eq 'C:\target\' }
+      Mock Test-Path { $false } -ParameterFilter { $path -eq 'C:\target_tmp' }
+      Mock Get-ChildItem
+      Mock Move-Item
+      Mock Remove-Item
+
+      Add-DscResourceModule -Name SomeDSCModule -Path 'C:\target\' -RequiredVersion '1.0.0'
+
+      It 'downloads the latest version of the module' {
+        Assert-VerifiableMock
+        Assert-MockCalled New-Item -Times 1 -ParameterFilter { ($path -eq 'C:\target\') -and ($ItemType -eq 'Directory') }
+        Assert-MockCalled New-Item -Times 1 -ParameterFilter { ($path -eq 'C:\target_tmp') -and ($ItemType -eq 'Directory') }
+        Assert-MockCalled Save-Module -Times 1 -Debug -ParameterFilter {
+          ($Name -eq 'SomeDSCModule') `
+            -and ($path -eq 'C:\target_tmp') `
+            -and ($RequiredVersion -eq '1.0.0')
+        }
+        Assert-MockCalled Remove-Item -Times 1 -ParameterFilter { $path -eq 'C:\target_tmp' }
+      }
+
+      It 'does not leave litter behind' {
```

Comment:
Can this test fail if we mock `New-Item`?

Reply:
removed this test by using the testdrive :crossed_fingers: